backend/rates/rates.go

```diff
@@ -200,13 +204,21 @@ func (updater *RateUpdater) HistoricalPriceAt(coin, fiat string, at time.Time) f
 		return 0 // no data
 	}
 	if data[idx].timestamp.Equal(at) {
-		return data[idx].value // don't need to interpolate
+		value := data[idx].value
```

Comment:
Instead of considering sats as a special case here, I think it would be better to handle it inside history.go:fetchGeckoMarketRange so that it's saved with the correct value in the cache db.

Reply:
I moved the multiplication for `sat` to `fetchGeckoMarketRange`